components/collection/drop/Generative.vue

```diff
@@ -1,61 +1,21 @@
 <template>
-  <div class="unlockable-container">
-    <div class="container is-fluid border-top pt-6">
-      <div class="columns is-desktop">
-        <div class="column is-half-desktop mobile-padding">
-          <UnlockableCollectionInfo
-            :collection-id="collectionId"
-            :description="description" />
-          <hr />
-
-          <UnlockableTag :collection-id="collectionId" />
-
-          <CollectionDropMintSection
-            v-if="!isMobile"
-            :has-user-minted="hasUserMinted"
-            :is-wallet-connecting="isWalletConnecting"
-            :is-image-fetching="isImageFetching"
-            :minimum-funds="minimumFunds"
-            :minimum-funds-description="minimumFundsDescription"
-            :max-count="maxCount"
-            :minted-count="mintedCount"
-            :mint-count-available="mintCountAvailable"
-            :disabled="mintButtonDisabled"
-            :mint-button-label="mintButtonLabel"
-            @mint="handleSubmitMint" />
-        </div>
-
-        <div class="column pt-5 is-flex is-justify-content-center">
-          <GenerativePreview
-            :content="drop.content"
-            :image="drop.image"
-            @select="handleSelectImage" />
-        </div>
-
-        <CollectionDropMintSection
-          v-if="isMobile"
-          class="column"
-          :has-user-minted="hasUserMinted"
-          :is-wallet-connecting="isWalletConnecting"
-          :is-image-fetching="isImageFetching"
-          :minimum-funds="minimumFunds"
-          :minimum-funds-description="minimumFundsDescription"
-          :max-count="maxCount"
-          :minted-count="mintedCount"
-          :mint-count-available="mintCountAvailable"
-          :disabled="mintButtonDisabled"
-          :mint-button-label="mintButtonLabel"
-          @mint="handleSubmitMint" />
-      </div>
-
-      <CollectionUnlockableItemInfo :collection-id="collectionId" />
-      <div class="my-4">
-        <CarouselTypeLatestMints
-          :collection-id="collectionId"
-          interaction="MINT" />
-      </div>
-    </div>
-  </div>
+  <CollectionDropGenerativeLayout
+    :has-user-minted="hasUserMinted"
+    :is-wallet-connecting="isWalletConnecting"
+    :is-image-fetching="isImageFetching"
+    :is-loading="isLoading"
+    :minimum-funds="minimumFunds"
+    :minimum-funds-description="minimumFundsDescription"
+    :max-count="maxCount"
+    :minted-count="mintedCount"
+    :mint-count-available="mintCountAvailable"
+    :mint-button-label="mintButtonLabel"
+    :mint-button-disabled="mintButtonDisabled"
```

Comment:
it's a bit unusual for componnet to have so many props
do you think it can be improved?

Reply:
I agree that this component has a lot of props, we've removed some duplication don't want to split it into Mutiple components since they are identical. 

how would you do this ?

Reply:
normally things like like label or disabled etc are internally calculated
 not sure it's applicable here
 
 another option is to groups related props together for example  we could have `mintButtonProps`

Reply:
or maybe introduce some router component?